tests/test_app_text.py

```diff
@@ -114,7 +113,9 @@ def test_asset_variable_replacement(self):
                                        asset=invalid_asset,
                                        variables=test_vars)
         error_key = resource.asset.split()[-1]
-        self.assertEqual(error_key, "u'variable'")
+        if (error_key.startswith("u")):
```

Comment:
I believe I changed that because of the way unicode strings are handled in py2 vs py3. In py2, the result of this test is as "u'variable'" whereas in py3, it is "'variable'" (but they are both unicode strings for 'variable'), so I added that line to account for both. It's probably not the best way to do that, though - is there a cleaner way?

Reply:
[This looks like a better way](https://stackoverflow.com/questions/9079036/detect-python-version-at-runtime/9079062#9079062), thanks for the explanation!